src/prefect/agent/agent.py

```diff
@@ -69,16 +70,23 @@ class Agent:
             Agents when polling for work
         - env_vars (dict, optional): a dictionary of environment variables and values that will be set
             on each flow run that this agent submits for execution
+        - max_polls (int, optional): maximum number of times the agent will poll Prefect Cloud for flow runs;
+            defaults to infinite
     """
 
     def __init__(
-        self, name: str = None, labels: Iterable[str] = None, env_vars: dict = None
+        self,
+        name: str = None,
+        labels: Iterable[str] = None,
+        env_vars: dict = None,
+        max_polls: int = None,
```

Comment:
Not a big deal either way, but because `math.inf` is a float you could safely use it as the default option here instead of `None`

Reply:
The reason why I opted to use None was for the `int` typing